samples/drivers/led_strip/boards/stamp_c3.overlay

```diff
@@ -0,0 +1,46 @@
+/*
+ * Copyright (c) 2024 Croxel, Inc.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+#include <zephyr/dt-bindings/gpio/gpio.h>
+#include <zephyr/dt-bindings/led/led.h>
+
+#include <zephyr/dt-bindings/led/led.h>
+
+&uart0 {
+	status = "okay";
+	current-speed = <2500000>;
+	pinctrl-0 = <&uart0_default>;
+	pinctrl-names = "default";
+	data-bits = <7>;
+    tx-invert;
```

Comment:
Is that needed to have a "low -> high" transition for the start bit (instead of "low -> high") ? And to use it as the start of a data bit sent over the wire (and vice versa with the stop bit) ?

Is not `tx-invert` a property specific to STM32 UART controllers ? If yes, is this driver working with other UART controllers ?

And also the indentation above is wrong :)

Reply:
this seems far too limited for widespread use, I haven't got any mcus with a tx invert and seemingly neither does the mcu you are using this on

Reply:
@simonguinot Yes, it is needed because in the default UART TX line, the ideal state is high. This could be wrongly assumed to be some data by the WS2812 LED or similar ICs.
Many MCUs support UART line inversion with their HAL libraries. While writing this driver, I could only find inversion support for STM32 in Zephyr. That's why I created a separate PR [#72574](https://github.com/zephyrproject-rtos/zephyr/pull/72574/commits/fe7e05138ada5a71989bd07b35e678dac7a38e49) for supporting Esp32 uart pin inversion. NXP also has their PR [#71802](https://github.com/zephyrproject-rtos/zephyr/pull/71802) for supporting uart pin inversion merged into zephyr.

Reply:
@thedjnK The SPI and I2s peripherals are limited in smaller microcontrollers thats why support for uart is a good addition. Most of the mcu vensors supports uart pins inversion natively and their supports is getting added into zephyr day by day. As of now support for STM32, NXP and ESP32 mcu has already been added. And as mentioned in the description, I have used esp32c3 mcu to for testing it was working fine on my end.

Reply:
Therefore this property is mandatory. So the dependency must be expressed somewhere. If the `tx-invert` property can be made generic, then maybe you can use a DT macro in the driver's Kconfig (such as `dt_node_has_prop`) to allow the driver to be enabled only if this property is available. Or something else... But the dependency must be expressed somewhere.

In addition, you have to explain it in the driver's DT binding. For now, it is written:

```
The Uart driver should be usable as long as a zephyr Uart API driver
is available for your board. Hardware specific tuning is required
using these properties:

  - uart-baudrate
```

It is not true. Without inverting the Tx levels, the driver won't work.

I'm also waiting for a detailed explanation in the binding on how the data is serialized over the Tx UART wire by using the start/stop bits with the signal polarity inverted. And a summarized description in the commit message would also be welcome :)

Reply:
Yeah, it make sense to mention that this property is mandatory right now. Okay will add this.
Also, as you requested I added the comments explaining serialization in the ws2812_uart.c files. 
Explaination:

3 pulses/bits of UART are used to represent one bit of RGB data. During the development of this driver, I was using an ESP32C3 MCU with its UART baud rate set to 2.5MHz. Hence, 1 bit equals 400ns.
 For WS2812 timing details, please refer to its datasheet. 
Link: https://cdn-shop.adafruit.com/datasheets/WS2812.pdf
We can represent logic 1 with 0b110 (first two bits high,remaining one bit low), and logic 0 with 0b100 (first bit high, remaining two bits low).
 We also need to invert the polarity of UART pins because in default, the idle TX line is high, but WS2812 or similar LEDs need the idle line to be low, otherwise, it will misinterpret.
 When we make the TX line inverted, the start bit will be low to high and the stop bit will be high to low.
 Hence, the nine bits we are sending in one go will always have 1 at the start and 0 at the end, we just need to calculate middle 7 bits.
 Also, keep in mind that UART always sends the MSB first. Keeping all this data in mind, I created a lookup table to ease the calculations.
 This function is just taking three bits at a time from the 24-bit color data, i.e., color[3] array (MSB first),
 and passing those three bits into the lookup table ws2812_uart_lut[] to  retrieve the 7-bit data and fills out
the serialized output buffer.

Reply:
> If the tx-invert property can be made generic, then maybe you can use a DT macro in the driver's Kconfig (such as dt_node_has_prop) to allow the driver to be enabled only if this property is available

@AyushKot96 as it was suggested here: https://github.com/zephyrproject-rtos/zephyr/pull/72574#discussion_r1598880697, I think it makes sense to follow-up with a PR to generalize this property and then enforce it on this driver.

Reply:
@simonguinot I think having internal pin inversion capability facilitate direct connection of led with uart tx pin. but if internal is not supported then external logic inversion circuit can be used. thus we should not make it mandatory instead mention it in the documentation.

In any case, here is the PR to make tx/rx generic properties of uart driver as mentioned in  https://github.com/zephyrproject-rtos/zephyr/pull/72574#discussion_r1598880697 https://github.com/zephyrproject-rtos/zephyr/pull/72947

CC: @ubieda

Reply:
@cx-anuj-pathak external logic inversion of the UART Tx pin also needs to be described in DT somehow, probably using the tx-invert property as well...

Anyway, from the LED point of view, we need a dependency to ensure that the ws2812_uart driver can be used. And I think it would be smart to use the `tx-invert` property. Else @AyushKot96 will have to find another way.

Reply:
@AyushKot96 @cx-anuj-pathak I suggest keeping this PR simple and scoping this first version of the driver to have the `tx-invert` property as a requirement to work.

Reply:
@ubieda @simonguinot tx-invert property is now a requirement for this driver to work.